packages/fluentui/react-bindings/src/styles/resolveStyles.ts

```diff
@@ -43,20 +43,48 @@ const resolveStyles = (
   resolvedVariables: ComponentVariablesObject,
   renderStylesInput?: (styles: ICSSInJSStyle) => string
 ): ResolveStylesResult => {
-  const { className: componentClassName, theme, displayName, props, rtl, disableAnimations, renderer, performance } = options || {};
+  const { className: componentClassName, theme, displayName, props, rtl, disableAnimations, renderer, performance } = options;
 
   const { className, design, styles, variables, ...stylesProps } = props;
-  const noInlineOverrides = !(design || styles || variables);
 
-  const cacheEnabled = performance.enableStylesCaching && noInlineOverrides;
+  const noInlineStylesOverrides = !(design || styles);
+  const noVariableOverrides = performance.enableHardVariablesCaching || !variables;
+
+  /* istanbul ignore else */
+  if (process.env.NODE_ENV !== 'production') {
+    if (!performance.enableStylesCaching && performance.enableHardVariablesCaching) {
+      throw new Error(
+        '@fluentui/react: Please check your "performance" settings on "Provider", to enable "enableHardVariablesCaching" you need to enable "enableStylesCaching"'
+      );
+    }
+
+    if (performance.enableHardVariablesCaching && variables) {
+      if (!_.isPlainObject(variables)) {
+        throw new Error('@fluentui/react: With "enableHardVariablesCaching" only plain objects can be passed to "variables" prop.');
+      }
+
+      const hasOnlyBooleanVariables = Object.keys(variables).every(variableName => {
+        return (
+          variables[variableName] === null || typeof variables[variableName] === 'boolean' || typeof variables[variableName] === 'undefined'
+        );
+      });
+
+      if (!hasOnlyBooleanVariables) {
```

Comment:
We should not throw here, we should just not cache for this instance. We are not throwing in the other cases of caching, so we shouldn't here too. We cannot guarantee that **all** usages of variables are booleans in a project.

Reply:
During offline conversation we discovered issues and I changed the implementation to avoid throwing errors.